cmd/kube_create.go

```diff
@@ -102,12 +106,20 @@ curl -fsSL https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key
 cat <<EOF | sudo tee /etc/apt/sources.list.d/kubernetes.list
 deb http://apt.kubernetes.io/ kubernetes-xenial main
 EOF
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get update
 
-sudo -E DEBIAN_FRONTEND=noninteractive apt-get update && \
-sudo -E DEBIAN_FRONTEND=noninteractive apt-get install -y kubelet kubeadm kubectl && \
+export PKG_VERSION=$(apt-cache madison kubelet | awk '$3 ~ /{{ .Version }}/ { print $3 }' | head -n 1)
+if [[ -z "${PKG_VERSION}" ]]; then
+	echo "error: unable to find package for version {{ .Version }}" >&2
+	exit 1
+fi
+
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get install -y kubelet=${PKG_VERSION} \
+	kubeadm=${PKG_VERSION} \
+	kubectl=${PKG_VERSION} && \
```

Comment:
what's the risk that the `PKG_VERSION` of `kubelet` isn't the same as the one of `kubeadm` or `kubectl`?

Reply:
With this code I'd say `PKG_VERSION` will be the same for all `kubelet`, `kubeadm` and `kubectl`. There may be a problem with the `kubectl` version that the user is supposed to install locally once the cluster instance is bootstrapped, though.